icic_extraction/grid.py

```diff
@@ -34,10 +43,14 @@ def extract_transaction_from_detailed_table(detailed_table):
     """
     transactions=[]   
     for record in detailed_table :
-        records_list = record.split()
-        name = " ".join(records_list[5:len(records_list)-7])
-        del records_list[5:len(records_list)-7]
-        records_list.insert(5,name)
+        record_clean = re.sub("NSE","",record)
```

Comment:
this is too generic, if we have a company that has NSE in its name, it will be removed 
another solution is to remove it from the end of the string